x/auth/keeper/keeper.go

```diff
@@ -275,3 +278,43 @@ func (ak AccountKeeper) GetParams(ctx context.Context) (params types.Params) {
 	}
 	return params
 }
+
+func (ak AccountKeeper) NonAtomicMsgsExec(ctx context.Context, signer sdk.AccAddress, msgs []sdk.Msg) ([]*types.NonAtomicExecResult, error) {
+	var msgsAny []*codectypes.Any
+	msgResponses := make([]*types.NonAtomicExecResult, 0, len(msgs))
+
+	for _, msg := range msgs {
+		if m, ok := msg.(sdk.HasValidateBasic); ok {
+			if err := m.ValidateBasic(); err != nil {
+				value := &types.NonAtomicExecResult{Error: err.Error()}
+				msgResponses = append(msgResponses, value)
+				continue
+			}
+		}
+
+		anyValue, err := codectypes.NewAnyWithValue(msg)
+		if err != nil {
+			value := &types.NonAtomicExecResult{Error: err.Error()}
+			msgResponses = append(msgResponses, value)
+			continue
+		}
+		msgsAny = append(msgsAny, anyValue)
+
+	}
+
+	results, err := ak.AccountsModKeeper.SendAnyMessages(ctx, signer, msgsAny)
```

Comment:
this `MUST` be executed using the branch.Service, we cannot allow the state machine to be left in half state (eg a staking delegation half-failing, half succeeding)